x-pack/plugins/cloud_security_posture/public/pages/findings/latest_findings_by_resource/use_findings_by_resource.ts

```diff
@@ -22,77 +33,99 @@ export type CspFindingsByResourceResult = FindingsQueryResult<
   unknown
 >;
 
-interface FindingsByResourceAggs extends estypes.AggregationsCompositeAggregate {
-  groupBy: {
-    buckets: FindingsAggBucket[];
-  };
+interface FindingsByResourceAggs {
+  resources: estypes.AggregationsMultiBucketAggregateBase<FindingsAggBucket>;
+  resource_total: estypes.AggregationsCardinalityAggregate;
 }
 
-interface FindingsAggBucket {
-  doc_count: number;
-  failed_findings: { doc_count: number };
-  key: {
-    resource_id: string;
-    cluster_id: string;
-    cis_section: string;
-  };
+interface FindingsAggBucket extends estypes.AggregationsStringRareTermsBucketKeys {
+  failed_findings: estypes.AggregationsMultiBucketBase;
+  cis_sections: estypes.AggregationsMultiBucketAggregateBase<estypes.AggregationsStringRareTermsBucketKeys>;
+  name: estypes.AggregationsMultiBucketAggregateBase<estypes.AggregationsStringRareTermsBucketKeys>;
+  type: estypes.AggregationsMultiBucketAggregateBase<estypes.AggregationsStringRareTermsBucketKeys>;
 }
 
 export const getFindingsByResourceAggQuery = ({
   index,
   query,
-}: FindingsBaseEsQuery): estypes.SearchRequest => ({
+  from,
+  size,
+}: UseResourceFindingsOptions): estypes.SearchRequest => ({
   index,
-  size: 0,
   body: {
     query,
+    size: 0,
     aggs: {
-      groupBy: {
-        composite: {
-          size: 10 * 1000,
-          sources: [
-            { resource_id: { terms: { field: 'resource_id.keyword' } } },
-            { cluster_id: { terms: { field: 'cluster_id.keyword' } } },
-            { cis_section: { terms: { field: 'rule.section.keyword' } } },
-          ],
-        },
+      resource_total: { cardinality: { field: 'resource.id.keyword' } },
+      resources: {
+        terms: { field: 'resource.id.keyword', size: 1000000 },
         aggs: {
+          cis_sections: {
+            terms: { field: 'rule.section.keyword', size, order: { _key: 'asc' } },
+          },
+          name: {
+            terms: { field: 'resource.name.keyword', size: 1, order: { _key: 'asc' } },
+          },
+          type: {
+            terms: { field: 'resource.sub_type.keyword', size: 1, order: { _key: 'asc' } },
+          },
           failed_findings: {
             filter: { term: { 'result.evaluation.keyword': 'failed' } },
           },
+          sort_failed_findings: {
+            bucket_sort: {
+              from,
+              size,
```

Comment:
`size` and `from` are both from `estypes.SearchRequest` and are indeed used for pagination. kept the naming to avoid introducing new terminology for existing terms. 

i didn't want to have this function take `pageIndex` and `pageSize` because that mixes 2 types of params: `eui/kibana` and `elasticsearch`. now the hook only knows `elasticsearch` and the ui only knows `eui/kibana` params. when this hook is called, we map. the `pageIndex` and `pageSize` from the query to `from` and `size`